bloom-housing/bloom
# feat: creating rate limiting

## Issue Overview

This PR addresses [#issue](https://github.com/bloom-housing/bloom/issues/3950)


## Description
This should create a rate limiter across all of our endpoints! **There are still some logs in there since I need to see how this functions when its passed through the proxies in a hosted instance**

## How Can This Be Tested/Reviewed?
You'll need to set the new env variables locally. To make testing easier set THROTTLE_LIMIT to be like (like < 10)

Easiest way to test this is to hit: http://localhost:3101/api/#/root/healthCheck
and just hit it a few times, you should be good up until you cross THROTTLE_LIMIT

then it'll error


## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
